### DISCUSSION TITLE
Is it possible to skip page generation in getStaticProps?

### DISCUSSION THREAD

**Discussion by chooblarin** [vercel/next.js#20355] [Help]
I wanted to skip SSG in some cases, and found `notFound: true` option ([added in Next.js 10](https://nextjs.org/blog/next-10#redirect-and-notfound-support-for-getstaticprops--getserversideprops)). 

But when I tried the option, an error occur. 

Here is my minimal code. 

```js
// pages/posts.js
export default function Posts() {
  return <div>posts page</div>;
}

export const getStaticProps = () => {
  // In my case, it needs to return `notFound: true` conditionally. 
  return { notFound: true };
};
```

and run `next build && next export`, then got the above error

```
{ [Error: ENOENT: no such file or directory, copyfile 'xxx/static-not-found/.next/server/pages/posts.html' -> 'xxx/static-not-found/out/posts.html']
  errno: -2,
  code: 'ENOENT',
  syscall: 'copyfile',
  path:
   'xxx/static-not-found/.next/server/pages/posts.html',
  dest:
   'xxx/static-not-found/out/posts.html' }
```

I don't want `posts.html` to be exported if `notFound: true` passed. 
Is it possible to skip page generation in getStaticProps? Is there anything I've missed...?

**Comment by fracture91** [comment]
@chooblarin I've filed a bug report for this: https://github.com/vercel/next.js/issues/24422

I was able to work around the problem by [writing empty html/json files where `next export` expects them to be](https://github.com/fracture91/ahurle-dev/blob/83922ad222cce787fec5c222182253b8e50c63ac/helpers/getBlogStaticProps.ts#L13-L41), but this is not ideal since users will get a 200 response with a blank page rather than the usual 404 behavior

**Comment by chooblarin** [comment]
It seems https://github.com/vercel/next.js/pull/24481 fixed my problem.
